src/java.desktop/share/classes/java/awt/Robot.java

```diff
@@ -218,7 +220,9 @@ private static void checkIsScreenDevice(GraphicsDevice device) {
      * @param y         Y position
      */
     public synchronized void mouseMove(int x, int y) {
-        peer.mouseMove(x, y);
+//        peer.mouseMove(x, y);
+        peer.mouseMove(Math.min(Math.max(x, screenBounds.x), screenBounds.x + screenBounds.width),
+                Math.min(Math.max(y, screenBounds.y), screenBounds.y + screenBounds.height));
         afterEvent();
```

Comment:
@alisenchung It might be good to add a simple test case. Probably robot.mouseMove() to off-screen coordinates and pass the test if the coordinates returned by MouseInfo.getPointerInfo().getLocation() is clamped (limited to screen bounds) else fail.

Reply:
oops, i had a test and forgot to include it in the PR, good reminder!